worker/src/main/java/gov/cms/ab2d/worker/processor/PatientClaimsProcessorImpl.java

```diff
@@ -39,27 +48,109 @@
     @Value("${bfd.earliest.data.date:01/01/2020}")
     private String earliestDataDate;
 
+    @Value("${aggregator.directory.finished}")
+    private String finishedDir;
+
+    @Value("${aggregator.directory.streaming}")
+    private String streamingDir;
+
+    @Value("${aggregator.file.buffer.size}")
+    private int bufferSize;
+
     private static final OffsetDateTime START_CHECK = OffsetDateTime.parse(SINCE_EARLIEST_DATE, ISO_DATE_TIME);
 
     /**
      * Process the retrieval of patient explanation of benefit objects and return the result
      * for further post-processing
      */
-    @Trace(metricName = "EOBRequest", dispatcher = true)
     @Async("patientProcessorThreadPool")
-    public Future<EobSearchResult> process(PatientClaimsRequest request) {
+    public Future<ProgressTrackerUpdate> process(PatientClaimsRequest request) {
+        ProgressTrackerUpdate update = new ProgressTrackerUpdate();
         final Token token = request.getToken();
         token.link();
-
+        FhirVersion fhirVersion = request.getVersion();
         try {
-            List<IBaseResource> eobs = getEobBundleResources(request);
-            EobSearchResult result = new EobSearchResult(request.getJob(), request.getContractNum(), eobs);
-            return new AsyncResult<>(result);
+            String anyErrors = writeOutData(request, fhirVersion, update);
+            if (anyErrors != null && anyErrors.length() > 0) {
+                writeOutErrors(anyErrors, request, fhirVersion);
+            }
         } catch (Exception ex) {
             return AsyncResult.forExecutionException(ex);
         } finally {
             token.expire();
         }
+        return AsyncResult.forValue(update);
+    }
+
+    private String writeOutData(PatientClaimsRequest request, FhirVersion fhirVersion, ProgressTrackerUpdate update) throws IOException {
+        File file = null;
```

Comment:
Why not write out the errors when they happen instead of at the very end?

Reply:
Wanted each stream to have its own try with resources black and not have them nested.